### PR TITLE
Updated language to address concerns around null (Fixes #14074)

### LINKED ISSUE #14074: Slightly dirty explanation of null.Equals(null)
The top of the guide has the following statement:
5.  x.Equals(null) returns false. However, null.Equals(null) throws an exception; it does not obey rule number two above.

This suffers from a few problems:

- `null.Equals(null)` is an example of a violation of rule 1.  An example of a violation of rule 2 is `null.Equals(x)`.
- `null.Equals(null)` won't even compile, unless the left side accessed via variable or is cast before usage.  As such, a beginner user attempting to try this example themselves may have a different experience than the guide predicts, but won't necessarily understand why.
- Personally, I feel like the rules would be better described as the "four guarantees of equivalence." Rule five is more of an exception to these rules than part of them.

As an alternative, we could add bullet points immediately below the guarantees:

- If `x` is `null`, `x.Equals(x)` throws an exception.  Rule number one does not apply.
- If `x` is `null`, `x.Equals(y)` throws an exception, even if y is `null`.  Rule number two does not apply.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e95b738e-a319-8947-c098-692068c24b9c
* Version Independent ID: 227a6999-9add-4a0a-98cb-c0c6624d6089
* Content: [How to: define value equality for a type - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type#feedback)
* Content Source: [docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by IntegerMan**
## Summary

Added disclaimer parenthetical around nulls along with a paragraph break to delineate the context shift to examples.

Reworded rule 5 per Bill's wording. It is more technical than the rest of the page, however I found value in the technicality of the explanation regarding the CLR and this reference.

Fixes #14074

### CODE REVIEW

File: docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md [hunk 1]
```diff
@@ -61,5 +63,6 @@ When you define a class or struct, you decide whether it makes sense to create a
   
 ## See also
 
-- [Equality comparisons](equality-comparisons.md)
-- [C# Programming Guide](../index.md)
+-[Equality comparisons](equality-comparisons.md)
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md]: It was correct with the space.

```suggestion
- [Equality comparisons](equality-comparisons.md)
```

File: docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md [hunk 2]
```diff
@@ -61,5 +63,6 @@ When you define a class or struct, you decide whether it makes sense to create a
...
+-[C# Programming Guide](../index.md)
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md]:
```suggestion
- [C# Programming Guide](../index.md)
```

File: docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md [hunk 3]
```diff
@@ -61,5 +63,6 @@ When you define a class or struct, you decide whether it makes sense to create a
...
+
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md]: nit: Remove extra space

```suggestion

```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

File: docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md (hunk 3 above)
Comment (IntegerMan) [reply docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md]: @Youssef1313 the way it was seems to have been triggering the linter failure as the lint now fails. I'm not sure I understand what's wrong there.

File: docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md [hunk 4]
```diff
@@ -62,4 +64,5 @@ When you define a class or struct, you decide whether it makes sense to create a
 ## See also
 
 - [Equality comparisons](equality-comparisons.md)
-- [C# Programming Guide](../index.md)
+- [C# programming guide](../index.md)
+
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type.md]: This one should fix it.

```suggestion
```

**Review by BillWagner — APPROVED** [review]
This is also great @IntegerMan 

I'll :shipit: now.

Thanks again for contributing to docs! 🎆

**Pull request merged**